eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -195,18 +194,24 @@ class PeerConnection(keyPair: KeyPair, conf: PeerConnection.Conf, switchboard: A
           d.transport ! Pong(ByteVector.fill(pongLength)(0.toByte))
         } else {
           log.warning(s"ignoring invalid ping with pongLength=${ping.pongLength}")
+          d.transport ! Warning(s"invalid pong length (${ping.pongLength})")
         }
         stay
 
       case Event(pong@Pong(data), d: ConnectedData) =>
         d.transport ! TransportHandler.ReadAck(pong)
         d.expectedPong_opt match {
-          case Some(ExpectedPong(ping, timestamp)) if ping.pongLength == data.length =>
-            // we use the pong size to correlate between pings and pongs
-            val latency = System.currentTimeMillis - timestamp
-            log.debug(s"received pong with latency=$latency")
-            cancelTimer(PingTimeout.toString())
-          // we don't need to call scheduleNextPing here, the next ping was already scheduled when we received that pong
+          case Some(ExpectedPong(ping, timestamp)) =>
+            if (ping.pongLength == data.length) {
+              // we use the pong size to correlate between pings and pongs
+              val latency = System.currentTimeMillis - timestamp
+              log.debug(s"received pong with latency=$latency")
+              cancelTimer(PingTimeout.toString())
+              // we don't need to call scheduleNextPing here, the next ping was already scheduled when we received that pong
+            } else {
+              log.warning(s"ignoring invalid pong with length=${data.length}")
+              d.transport ! Warning(s"invalid pong length (${data.length})")
+            }
           case None =>
             log.debug(s"received unexpected pong with size=${data.length}")
```

Comment:
Maybe we could send a warning for unexpected pongs too? Or you could have factored all cases in the `case None`, sending a generic "invalid pong with length x" ?

Reply:
Done in https://github.com/ACINQ/eclair/pull/1863/commits/bf10f462f50cd45c865e59370b572a42f736cd39